libdnf/base/transaction.cpp

```diff
@@ -372,6 +372,34 @@ std::string Transaction::all_package_solver_problems_to_string() {
     return output;
 }
 
+void Transaction::Impl::set_transaction_internal(
```

Comment:
This method name is very bad. I have an idea how to refactor this a bit differently though, but instead of trying to explain, I'll give it a try myself and push a fixup commit, after we decide what to do about the `LogicError`.